pkg/volume/rbd/rbd.go

```diff
@@ -235,3 +239,41 @@ func (plugin *rbdPlugin) execCommand(command string, args []string) ([]byte, err
 	cmd := plugin.exe.Command(command, args...)
 	return cmd.CombinedOutput()
 }
+
+type rbdRecycler struct {
+	name    string
+	spec    *volume.Spec
+	config  volume.VolumeConfig
+	host    volume.VolumeHost
+	timeout int64
+	volume.MetricsNil
+}
+
+func (r *rbdPlugin) NewRecycler(spec *volume.Spec) (volume.Recycler, error) {
+	return r.newRecyclerFunc(spec, r.host, r.config)
+}
+
+func newRecycler(spec *volume.Spec, host volume.VolumeHost, volumeConfig volume.VolumeConfig) (volume.Recycler, error) {
+	if spec.PersistentVolume == nil || spec.PersistentVolume.Spec.RBD == nil {
+		return nil, fmt.Errorf("spec.PersistentVolumeSource.rbd is nil")
+	}
+
+	return &rbdRecycler{
+		name:    spec.Name(),
+		spec:    spec,
+		host:    host,
+		config:  volumeConfig,
+		timeout: volume.CalculateTimeoutForVolume(volumeConfig.RecyclerMinimumTimeout, volumeConfig.RecyclerTimeoutIncrement, spec.PersistentVolume),
+	}, nil
+}
+
+func (r *rbdRecycler) GetPath() string {
+	return ""
+}
+
+func (r *rbdRecycler) Recycle() error {
+	pod := r.config.RecyclerPodTemplate
+	pod.GenerateName = "pv-recycler-rbd-"
+	pod.Spec.Volumes[0].VolumeSource = api.VolumeSource{RBD: r.spec.PersistentVolume.Spec.RBD}
```

Comment:
The code can be more defensive and ensure Volumes[0] exists.  That validation was added to the Origin version (https://github.com/openshift/origin/blob/master/pkg/cmd/server/kubernetes/master.go#L178).

Agree on docs and defensive coding for this.